### PR TITLE
[Key Vault] Update HSM deployment scripts

### PR DESCRIPTION
**Pull request by mccoyp**
HSM deployments have been stalling during live test pipelines for the past week. Discussion of the issue can be found in https://github.com/Azure/azure-sdk-for-python/pull/21717.

### CODE REVIEW

**Comment by mccoyp** [comment]
@benbp There are some `azure-keyvault-administration` errors that seem to have to do with SAS tokens, but this will unblock the issue with not being able to deploy HSMs

File: sdk/keyvault/test-resources.json [hunk 1]
```diff
@@ -37,7 +37,7 @@
         },
         "hsmLocation": {
             "type": "string",
-            "defaultValue": "westus",
+            "defaultValue": "canadacentral",
```
Comment (benbp) [inline sdk/keyvault/test-resources.json:40]: We will still pass in `westus2` as the default in the deployment scripts unless an override is specified in `tests.yml`

**Review by benbp — APPROVED** [review]

**Pull request merged**